--- setup.py	2022-12-25 14:10:24.000000000 +0100
+++ setup_changed.py	2022-12-25 14:11:28.000000000 +0100
@@ -58,7 +58,7 @@
     from setuptools import setup
 
     params = {
-        'install_requires': ['pyasn1>=0.1.8', 'pysmi', 'pycryptodome'],
+        'install_requires': ['pyasn1>=0.1.8', 'pysmi'],
         'zip_safe': True
     }
 
@@ -72,7 +72,7 @@
 
     params = {}
     if sys.version_info[:2] > (2, 4):
-        params['requires'] = ['pyasn1(>=0.1.8)', 'pysmi', 'pycryptodome']
+        params['requires'] = ['pyasn1(>=0.1.8)', 'pysmi']
 
 doclines = [x.strip() for x in (__doc__ or '').split('\n') if x]
 
