--- setup.py	2022-12-25 14:34:04.000000000 +0100
+++ setup.py	2022-12-25 14:34:30.000000000 +0100
@@ -53,8 +53,6 @@
         'install_requires': ['pyasn1>=0.1.8', 'pysmi'],
         'zip_safe': True
     }
-    if sys.platform.lower()[:3] != 'win':
-        params['install_requires'].append('pycrypto>=2.4.1')
 
 except ImportError:
     for arg in sys.argv:
@@ -66,8 +64,6 @@
     params = {}
     if sys.version_info[:2] > (2, 4):
         params['requires'] = ['pyasn1(>=0.1.8)', 'pysmi']
-        if sys.platform.lower()[:3] != 'win':
-            params['requires'].append('pycrypto(>=2.4.1)')
 
 if sys.platform.lower()[:3] == 'win':
     try:
